Azure/terraform/main.tf

```diff
@@ -2,14 +2,15 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "~> 3.9.0"
+      version = "~> 4.12.0"
     }
   }
 
   required_version = ">= 1.1.0"
 }
 
 provider "azurerm" {
+  skip_provider_registration = "true"
```

Comment:
I think it is better if we don't commit this setting and leave it up to the user to specify it.